### PR TITLE
Add support for grabbing the externalIP in the GCE cloud provider.

### PR DESCRIPTION
**Pull request by brendandburns**

### CODE REVIEW

File: pkg/cloudprovider/gce/gce.go [hunk 1]
```diff
@@ -125,6 +130,7 @@ func newGCECloud() (*GCECloud, error) {
 		projectID:        projectID,
 		zone:             zone,
 		instanceID:       instanceID,
+		metadataAccess:   getMetadata,
```
Comment (roberthbailey) [inline pkg/cloudprovider/gce/gce.go]: Was getMetadata unused before?

File: pkg/cloudprovider/gce/gce.go [hunk 2]
```diff
@@ -41,6 +41,8 @@ import (
 	"google.golang.org/cloud/compute/metadata"
 )
 
+const EXTERNAL_IP_METADATA_URL = "http://169.254.169.254/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip"
```
Comment (roberthbailey) [inline pkg/cloudprovider/gce/gce.go]: It seems like every URL passed into getMetadata will have the same prefix. It looks like this is the first one, but we will want to extract the common prefix out once we are making other requests to the metadata server.

File: pkg/cloudprovider/gce/gce.go (hunk 1 above)
Comment (brendandburns) [inline pkg/cloudprovider/gce/gce.go]: yes, it appears so.

File: pkg/cloudprovider/gce/gce.go (hunk 2 above)
Comment (brendandburns) [inline pkg/cloudprovider/gce/gce.go]: Can we do this when there is a second URL?

Comment (roberthbailey) [inline pkg/cloudprovider/gce/gce.go]: Yes.

**Comment by roberthbailey** [comment]
LGTM.

**Pull request merged**